### PR TITLE
Address Feedback + Autogen Docstring

### PR DESCRIPTION
**Pull request by Farhan-Faisal**
**Peer Feedback 1 <[htang085](https://github.com/htang085)>**. 

- [x] Test Edge Cases: Add tests for invalid inputs (e.g., missing API keys or malformed files). 
        - We added documentation for this. Integrating OpenAI itself was challenging, given the course load. We would make the tests for this more rigorous in future iterations.

**Peer Feedback 2 <[nhantien](https://github.com/nhantien)>**:  

- [x] I see you have a TODO to add OpenAI autogeneration, which is very creative, and it's also do-able with a combination of OpenAI python client + let the user set an environment variable for their API key.
      - Implemented

**Peer Feedback 3 <[Farhan-Faisal](https://github.com/Farhan-Faisal)>**
PEP8 Guidelines:  
- [x] In func_dto.py and generate_docstring_template, the function headers are too long. Consider making them multiline to adhere to PEP8.
     -  Headers broken down into multiline 

Refactoring: 
- [x] In func_dto.py, there is unnecessary use of list comprehension.
- [x] Instead of string parsing in func_dto.py, consider using inspect on the callable to retrieve type information directly. 

Documentation:  
- [x] In func_dto.py, class and method docstrings are missing. 
- [x] In the README, the output type of read_user_function is incorrect— it should be FunctionDTO.
- [x] In the read_user_function docstring, the output type is also incorrect— it should be FunctionDTO.
- [x] Include instructions for running tests with a coverage report in the README.
- [x] You might need people to set up OpenAI API key in their environments. This should be indicated in the README in future iterations.

Tests:  
- [x] Tests for read_user_function are failing. It seems like the type in the documentation is 'int', whereas it should be <class 'int'>.
- [x] Similar problem exists in test_fml_doc_gen
- [x] Docstring Completion (auto_generate):

Autogen:  
- [x] It would be great if docstring completion were working! Looking forward to that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,12 +38,16 @@ def square(base: int, pow: int) -> int:
 print(generate_docstring_template(square, output_file=None))
 ```
 
+## OpenAI Dependency:
+
+`fml_doc_gen` requires an OpenAI API key to generate docstrings. You can set your API key using the `OPENAI_API_KEY` environment variable. Check our our documentation [here](https://readthedocs.org/projects/fml-doc-gen/badge/?version=latest) for more information.
```
Comment (mikem2m) [inline README.md:43]: typo: check *out our

**Review by mikem2m — APPROVED** [review]
LGTM!
